### PR TITLE
Allow users to run Headstart on a site through Calypso.

### PR DESCRIPTION
**Pull request by kwight**
Headstart is the WordPress.com feature that sets up a new site on signup according to the selected theme. This gives a new user a great start towards having their "site look like the demo".

This PR would allow "Headstart on Demand": a user can choose to have Headstart run on one of their existing sites, aiding in setup any time a user switches themes.

Beginning mockup:

screen shot 2017-01-10 at 10 27 05 am

Previous discussion: p6nuNy-18d-p2

### CODE REVIEW

**Comment by melchoyce** [comment]
@michelleweber Can we get your feedback on copy here? ❤️

**Comment by michelleweber** [comment]
@melchoyce, of course. Here's a revision, in which I:
- Took the negative out of the first sentence.
- Took the hypen out of Setup -- it's not needed/correct in this usage.
- Took out the "Start from scratch" because I'm not sure how that internationalizes -- if you've determined that's not an issue, feel free to add it back!
- Tried to avoid the word "content" in favor of being more specific.
- Tried to explain the placeholder a bit more.

We can still probably improve this a little, but wanted to present a slightly different take. 

_Want your site to look like the demo? Use Auto Setup to automatically apply the demo site's settings to your site.

You can apply Auto Setup to your current site and keep all your posts, pages, and widgets, or use it for a fresh start and delete everything currently on your site. In both cases, placeholder text will appear on your site -- some themes need certain elements to look like the demo, so Auto Setup adds those for you. Please customize it!_

**Comment by melchoyce** [comment]
Thank you! 🙇 

@kwight — I'll get you some design feedback this afternoon or tomorrow morning.

**Comment by melchoyce** [comment]
Good use of `is-scary` on the Set Up From Scratch button. I agree with @kathrynwp that "Set Up and Keep Content" should be on the left, and I think we should probably add `is-primary` to that button to further call it out as the "safer," primary option. 

I'll hold off on more feedback until you have more of a chance to build it out. Want to ping me when you want me to review?

**Comment by kwight** [comment]
@michelleweber done, thanks! I've also started calling it "Theme Setup", instead of "Auto Setup" – thoughts?

File: client/my-sites/site-settings/theme-setup/index.jsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import { getActiveTheme, getTheme } from 'state/themes/selectors';
+import { isJetpackSite } from 'state/sites/selectors';
+import ActiveThemeScreenshot from './active-theme-screenshot';
+
+let ThemeSetup = ( { site, isJetpack, themeId, theme, translate, activeSiteDomain } ) => {
+	const onBack = () => {
+		page( '/settings/general/' + activeSiteDomain );
+	};
+
+	const noticeText = site && isJetpack ? translate( 'This feature is currently unavailable for Jetpack sites.' ) : translate( 'This action cannot be undone.' );
```
Comment (a8ci18n) [inline client/my-sites/site-settings/theme-setup/index.jsx]: Alternate string suggestion: 
* `translate( 'This cannot be undone' )`&mdash; translations: **3**. 

[PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/settings-theme-setup)

Comment (a8ci18n) [inline client/my-sites/site-settings/theme-setup/index.jsx]: Alternate string suggestion: 
* `translate( 'This cannot be undone' )`&mdash; translations: **3**. 

[PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/settings-theme-setup)

**Comment by kwight** [comment]
Patch for the Customizer text: https://mc.a8c.com/pb/17a3a/#diff

**Comment by kwight** [comment]
As of f1b085d. Patch is also done for the Customizer link (17a3a).

screen shot 2017-01-16 at 3 28 53 pm

screen shot 2017-01-16 at 3 33 21 pm

**Comment by folletto** [comment]
Design wise seems good. I'd personally try to avoid having the yellow notice bar inside a card, but I think here it works. :y: 

Tiny thing, can we add a border to the thumb image to avoid the whites to wash out on the white card? Like `border: 1px solid lighten( $gray, 30% );`

**Comment by melchoyce** [comment]
Yeah, was also thinking a border and maybe changing up the theme label styles a bit:

screen shot 2017-01-17 at 10 52 16 am

Copying the "CURRENT THEME" styling on https://calypso.live/design/:

screen shot 2017-01-17 at 10 54 22 am

**Comment by folletto** [comment]
Good one, yes 👍

File: client/my-sites/site-settings/theme-setup/index.jsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	return (
+		<div className="main theme-setup" role="main">
+			{ site && <QueryActiveTheme siteId={ site.ID } /> }
+			{ themeId && <QueryTheme siteId={ 'wpcom' } themeId={ themeId } /> }
+			<HeaderCake onClick={ onBack }><h1>{ translate( 'Theme Setup' ) }</h1></HeaderCake>
+			<ActionPanel>
+				<ActionPanelBody>
+					<ActionPanelTitle>{ translate( 'Theme Setup' ) }</ActionPanelTitle>
+					<Notice status={ 'is-warning' } showDismiss={ false }>
+						{ translate( 'This action cannot be undone.' ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/theme-setup/index.jsx]: Alternate string suggestion: 
* `translate( 'This cannot be undone' )`&mdash; translations: **3**. 

[PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/settings-theme-setup)

**Comment by kwight** [comment]
When loading the URL from a cold start (so no site info available yet, I've got a pulsing placeholder for the theme screenshot. But this means both the screenshot and the current site in the sidebar are going (oh, so much throbbing). @melchoyce @folletto Is there a better way design-wise to handle this?..

screen shot 2017-01-17 at 2 06 37 pm

**Comment by folletto** [comment]
> When loading the URL from a cold start (so no site info available yet), I've got a pulsing placeholder for the theme screenshot. But this means both the screenshot and the current site in the sidebar are going (oh, so much throbbing). @melchoyce @folletto Is there a better way design-wise to handle this?..

If I'm not mistaken, the entire UI is unusable until the site loads. In that case I'd just use the skeleton-loading animation on the _entire card_ instead of just the theme, as no action can be taken until that loads.

Something like:

  
screen shot 2017-01-18 at 12 49 48

**Comment by melchoyce** [comment]
Yeah, think we're just going to have to deal with excessive throbbing.

**Comment by kwight** [comment]
> If I'm not mistaken, the entire UI is unusable until the site loads. In that case I'd just use the skeleton-loading animation on the entire card instead of just the theme, as no action can be taken until that loads.

@folletto That seems like a lost opportunity to me. Even though the Theme Setup UI is in-actionable  until site/theme info arrives (buttons are disabled, no screenshot), at least the user can read the related text and see possible actions while waiting. Having even more throbbing with less info seems a step backward to me.

Of course, I'm happy to defer to design decisions, or if it's a Calypso consistency thing – just curious of the thinking behind it :)

**Comment by folletto** [comment]
It's just that skeletons usually apply to the entire block, and we tend to do the same in Calypso. I'm not entirely against it, it's just that for consistency I'd keep the loading always at card-level, not more granular than that.

File: client/my-sites/site-settings/theme-setup/theme-setup-card.jsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import Button from 'components/button';
+import ActiveThemeScreenshot from './active-theme-screenshot';
+
+const ThemeSetupCard = ( { theme, translate } ) => {
+	return (
+		<ActionPanel>
+			<ActionPanelBody>
+				<ActionPanelTitle>{ translate( 'Theme Setup' ) }</ActionPanelTitle>
+				<Notice status={ 'is-warning' } showDismiss={ false }>
+					{ translate( 'This action cannot be undone.' ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/theme-setup/theme-setup-card.jsx]: Alternate string suggestion: 
* `translate( 'This cannot be undone' )`&mdash; translations: **3**. 

[PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/settings-theme-setup)

**Comment by kwight** [comment]
New card-level skeleton. I just left the `p` blocks full-width, as getting the image in there too meant adding a bunch more CSS to behave properly at different breakpoints.

screen shot 2017-01-18 at 4 28 04 pm

**Comment by folletto** [comment]
> I just left the p blocks full-width, as getting the image in there too meant adding a bunch more CSS to behave properly at different breakpoints.

Good point. Simpler is better.

It's 👍  for me.

**Comment by melchoyce** [comment]
Looks good to me, too 👍

**Comment by kwight** [comment]
@michaeldcain noted that we should make it clear to the user that we're making live changes to the front-end (not just "trying out" or "previewing").

File: client/my-sites/site-settings/theme-setup/theme-setup-card.jsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+					} );
+			}
+		};
+
+		return (
+			<ActionPanel>
+				<ActionPanelBody>
+					<ActionPanelTitle>{ this.translate( 'Theme Setup' ) }</ActionPanelTitle>
+					<Notice status={ 'is-warning' } showDismiss={ false }>
+						{ this.translate( 'This action cannot be undone.' ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/theme-setup/theme-setup-card.jsx]: Alternate string suggestion: 
* `translate( 'This cannot be undone' )`&mdash; translations: **3**. 

[PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/settings-theme-setup)

**Comment by kwight** [comment]
@michelleweber hm, "Theme Setup", or "Site Setup"? I've been going with Theme Setup, but now I'm second-guessing myself.

**Comment by michelleweber** [comment]
+1 to Theme Setup. That's what it is; clarity and specificity FTW.

**Comment by kwight** [comment]
Replaced by:

#11762 
#11802 
#11803 
#11806 
#11812 
#11914 
#11925

**Pull request closed without merging**